Userland/Libraries/LibWeb/XHR/XMLHttpRequest.cpp

```diff
@@ -387,6 +387,12 @@ static bool is_forbidden_method(String const& method)
     return lowercase_method.is_one_of("connect", "trace", "track");
 }
 
+static bool is_method(String const& method)
+{
+    auto lowercase_method = method.to_lowercase();
+    return lowercase_method.is_one_of("delete", "get", "head", "options", "post", "put", "connect", "trace", "track");
```

Comment:
According to Fetch:
> A method is a [byte sequence](https://infra.spec.whatwg.org/#byte-sequence) that matches the [method](https://datatracker.ietf.org/doc/html/rfc7230#section-3.1.1) token production.

https://fetch.spec.whatwg.org/#concept-method

From the [RFC](https://datatracker.ietf.org/doc/html/rfc7230#section-3.1.1):
> The method token indicates the request method to be performed on the
   target resource.  The request method is case-sensitive.

>     method         = token

```
     token          = 1*tchar

     tchar          = "!" / "#" / "$" / "%" / "&" / "'" / "*"
                    / "+" / "-" / "." / "^" / "_" / "`" / "|" / "~"
                    / DIGIT / ALPHA
                    ; any VCHAR, except delimiters
```

This allows you to specify custom HTTP methods. For example:
![Screenshot from 2022-04-04 11-39-09](https://user-images.githubusercontent.com/25595356/161527358-ae686ef5-e29c-4acd-913b-d44952d71d69.png)

Reply:
Done as suggested, using regular expression